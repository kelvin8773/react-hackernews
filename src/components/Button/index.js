import React, { Component } from "react";

const Button = ({ onClick, className = "", children }) => {
  return (
    <button onClick={onClick} className={className} type="button">
      {children}
    </button>
  );
};

export default Button;
