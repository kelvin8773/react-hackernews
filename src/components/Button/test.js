import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Button from "./index";

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button>Give Me More!!</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a vaild Snapshot", () => {
    const component = renderer.create(<Button>Give Me More</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render children when pass in", () => {
    const element = shallow(<Button>More</Button>);
    expect(element.contains("More")).toBe(true);
  });
});
