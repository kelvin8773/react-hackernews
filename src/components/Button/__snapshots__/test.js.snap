// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a vaild Snapshot 1`] = `
<button
  className=""
  type="button"
>
  Give Me More
</button>
`;
